controllers/grid/settings/sections/SectionGridHandler.inc.php

```diff
@@ -202,19 +202,32 @@ function updateSection($args, $request) {
 	 */
 	function deleteSection($args, $request) {
 		$journal = $request->getJournal();
-
+		
 		$sectionDao = DAORegistry::getDAO('SectionDAO');
 		$section = $sectionDao->getById(
 			$request->getUserVar('sectionId'),
 			$journal->getId()
 		);
-
+		
 		if ($section && $request->checkCSRF()) {
-			$sectionDao->deleteObject($section);
-			return DAO::getDataChangedEvent($section->getId());
+			
+			$articleDao = DAORegistry::getDAO('ArticleDAO');
+			$checkSubmissions = $articleDao->retrieve('SELECT submission_id FROM submissions WHERE section_id = ?', (int) $request->getUserVar('sectionId'));
```

Comment:
We should maybe add the `journal_id` here. The `section_id` is unique regardless of context, but it's a useful sanity check to make sure that the section is within the context that's presumed to be editing at the moment.

Reply:
check